app/src/main/java/org/schabi/newpipe/views/NewPipeEditText.java

```diff
@@ -0,0 +1,69 @@
+package org.schabi.newpipe.views;
+
+import android.content.Context;
+import android.text.Selection;
+import android.text.Spannable;
+import android.util.AttributeSet;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.widget.AppCompatEditText;
+
+import org.schabi.newpipe.util.external_communication.ShareUtils;
+
+/**
+ * An {@link AppCompatEditText} which uses {@link ShareUtils#shareText(Context, String, String)}
+ * when sharing selected text by using the {@code Share} command of the floating actions.
+ * <p>
+ * This allows NewPipe to show Android share sheet instead of EMUI share sheet when sharing text
+ * from {@link AppCompatEditText} on EMUI devices.
+ * </p>
+ */
+public class NewPipeEditText extends AppCompatEditText {
+
+    public NewPipeEditText(@NonNull final Context context) {
+        super(context);
+    }
+
+    public NewPipeEditText(@NonNull final Context context, @Nullable final AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    public NewPipeEditText(@NonNull final Context context,
+                           @Nullable final AttributeSet attrs,
+                           final int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+    }
+
+    @Override
+    public boolean onTextContextMenuItem(final int id) {
+        final Spannable text = getText();
+        if (id == android.R.id.shareText) {
+            if (text != null) {
+                final String selectedText = getSelectedText(text).toString();
+                if (!selectedText.isEmpty()) {
+                    ShareUtils.shareText(getContext(), "", selectedText);
+                }
+                Selection.setSelection(text, getSelectionEnd());
```

Comment:
Why are you setting the selection again?

Reply:
I am setting the selection again to keep Android's behavoir when selecting text: after you press a option in the popup menu of the `TextView`/`EditText` (`Copy`, `Share`, ...), the text is not selected anymore. See [TextView's source code](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/widget/TextView.java#12929) for more details.